source/opt/inline_pass.cpp

```diff
@@ -402,23 +468,117 @@ bool InlinePass::Inline(ir::Function* func) {
   return modified;
 }
 
-bool InlinePass::IsInlinableFunction(const ir::Function* func) {
-  // We can only inline a function if it has blocks.
-  if (func->cbegin() == func->cend())
+bool InlinePass::hasMultipleReturns(ir::Function* func) {
+  bool seenReturn = false;
+  bool multipleReturns = false;
+  for (auto& blk : *func) {
+    auto lii = blk.cend();
+    lii--;
+    if (lii->opcode() == SpvOpReturn || lii->opcode() == SpvOpReturnValue) {
+      if (seenReturn) {
+        multipleReturns = true;
+        break;
+      }
+      seenReturn = true;
+    }
+  }
+  return multipleReturns;
+}
+
+void InlinePass::ComputeStructuredSuccessors(ir::Function* func) {
+  // If header, make merge block first successor.
+  for (auto& blk : *func) {
+    auto mii = blk.end();
+    mii--;
+    uint32_t mbid = 0;
+    if (mii != blk.begin()) {
```

Comment:
A vaild basic block always has at least two instructions: the label and the terminator instruction.
You're just being defensive here.  That's fine, but you should say so.

The practical impact is that for a valid module we'll never take the false branch.    That might show up one day in a code coverage report, so it will be nice to either (a)  make the invalid test case that proves you need this, or (b) add the comment.